src/Engine/System/RenderSystem.hpp

```diff
@@ -24,82 +25,72 @@
-    friend PostProcessShader;
-
-    public:
 
+public:
     static constexpr SystemID ID = SystemID::render;
 
-    public:
-
     static void init();
 
-    /* Iterate through shaders map
-        * Bind individual shaders 
-        * Call shaders' render function with the appropriate render component list */
+    /* Full render function including shadow maps, main render calls, and post-processing */
     static void update(float dt);
 
-    // creates a new shader and initializes it
-    template<typename ShaderT, typename... Args> static bool createShader(Args &&... args);
-
-    // takes possession of shader and initializes it
-    template <typename ShaderT> static bool addShader(UniquePtr<ShaderT> shader);
-
-    // get shader of the specified type
-    template <typename ShaderT> static ShaderT * getShader();
-
+    /* Camera */
     static void setCamera(const CameraComponent * camera);
-    
-    private:
+    static const CameraComponent * s_playerCamera;
+
+    /* Light */
+    static GameObject * s_lightObject;
+    static SpatialComponent * s_lightSpatial;
+    static CameraComponent * s_lightCamera;
+    static glm::vec3 getLightDir();
+    static void setLightDir(glm::vec3);
+    static float lightDist;
+
+    /* Shadows */
+    static const glm::mat4 & getL() { return s_shadowShader->getL(); }
+    static const GLuint getShadowMap() { return s_shadowShader->getShadowMapTexture(); }
+
+    /* Shaders */
+    template<typename ShaderT, typename... Args> static ShaderT * createShader(Args &&... args);
+    static DiffuseShader * s_diffuseShader;
+    static BounderShader * s_bounderShader;
+    static RayShader * s_rayShader;
+    static ShadowDepthShader * s_shadowShader;
+    static OctreeShader * s_octreeShader;
+    static PostProcessShader * s_postProcessShader;
+
+    /* FBO Stuff */
+    static GLuint getFBOTexture() { return s_fboColorTex; }
+
+private:
+    static Vector<DiffuseRenderComponent *> getFrustumComps(const CameraComponent *);
+    static const Vector<DiffuseRenderComponent *> & s_diffuseComponents;
 
-    static void initFBO();
 
+    static void initFBO();
     static void doResize();
-
-    private:
-
-    static const Vector<DiffuseRenderComponent *> & s_diffuseComponents;
-    static UnorderedMap<std::type_index, UniquePtr<Shader>> s_shaders;
-    static UniquePtr<PostProcessShader> s_postProcessShader;
-    static const CameraComponent * s_camera;
     static GLuint s_fbo;
     static GLuint s_fboColorTex;
     static bool s_wasResize;
 };
 
+
 // TEMPLATE IMPLEMENTATION /////////////////////////////////////////////////////
 
 template<typename ShaderT, typename... Args>
-bool RenderSystem::createShader(Args &&... args) {
-    auto shader(UniquePtr<ShaderT>::make(std::forward<Args>(args)...));
-    std::type_index typeI(typeid(ShaderT));
-
-    auto it(s_shaders.find(typeI));
-    if (it != s_shaders.end()) {
-        return true;
-    }
+ShaderT * RenderSystem::createShader(Args &&... args) {
```

Comment:
TODO : this is broken

Reply:
she shaders should prob just be initialized in rendersystem's init, instead of relying on main. I know it makes sense to do it in main because it uses file names, but it's the better option for the time we have.